lib/compile.js

```diff
@@ -130,7 +131,7 @@ var Compiler = (function () {
                     msgid = msgid.replace( unconvertedEntityPattern, convertedEntity );
                 }
 
-                if ( removeWhiteSpaces ) {
+                if ( removeWhiteSpaces && !isDBString ) {
```

Comment:
or rather use `this.shouldRemoveWhiteSpaces(item)`
where  `shouldRemoveWhiteSpaces` encapsulates conditions